### ISSUE TITLE
Fractional numbers for mobility months and days

### ISSUE THREAD

**Issue by mkurzydlowski** [erasmus-without-paper/ewp-specs-api-iias#23]
At UW we have fractional numbers in both average (total) months and average (total) days for student and staff mobilities respectively but avg(total)-days and avg(total)-months are defined as integers.

@erasmus-without-paper/all-members, could we allow fractional numbers in those fields?

**Comment by wrygiel** [comment]
I tried to find out if EUC issued any recommendations about this, but couldn't find any. In particular, our IIA template doesn't say anything about either allowing or disallowing fractional digits in "mobility numbers".

So, we are considering changing `xs:positiveInteger` to `xs:decimal` in 4 places in total.

Do you think we should also try to enforce some restrictions on the number of fractional digits? My guess is *no* - since EUC doesn't define any restrictions here, then we also shouldn't.

**Comment by wrygiel** [comment]
@georgschermann, could you tell us how you handle this in your database? How many fractional digits do you keep? Or, perhaps, you allow other rational numbers, e.g. "1/3"?

**Comment by georgschermann** [comment]
We keep 2 fractional digits on the average fields.

**Comment by wrygiel** [comment]
> We keep 2 fractional digits on the average fields.

I believe it's the same Poland does. I guess it's safe to assume this format for the APIs.

**Comment by wrygiel** [comment]
How do we update this? Note, that currently `<avg-months>` and `<total-months>` are both defined as required integers.

### Solution 1

Since both these elements are required, and **clients might have already assumed** that they're integers, we should not change them in a backward-incompatible way. The easiest and "the least ugly" solution I could find for that is adding a new optional attribute like this:

```xml
<avg-months exact-value="5.25">5</avg-months>
<avg-months exact-value="5.75">6</avg-months>
<avg-months exact-value="6.00">6</avg-months>
<avg-months>6</avg-months>
```

It works like this:

* Older clients will correctly parse the old "rounded" value (which in most cases will be equal to the exact value).
* We instruct the newer clients to check if `exact-value` exists. If it does then take the value from there. If it doesn't, then use the value from inside the element.
* Servers will need to produce the rounded values, if only for compatibility with older clients.

**Comment by wrygiel** [comment]
### Solution 2

Change the specs in a backward-incompatible way. I am against that, because the specification is marked as "stable", but some other developers asked me to include this solution here, because none of the clients are being used in production environment.

```xml
<avg-months>5.25</avg-months>
<avg-months>5.75</avg-months>
<avg-months>6.00</avg-months>
<avg-months>6</avg-months>
```

This change MAY break existing clients. We should get an approval from all partners (at least all those which have already implemented a client for IIAs) if we want to change the specs in this way.

**Comment by wrygiel** [comment]
@erasmus-without-paper/all-members (especially SOP and SIGMA, because they have indicated that they already implemented an IIA client) - which of these two solutions do you prefer?

**Comment by huntering** [comment]
solution 2 is ok for us.

**Comment by wrygiel** [comment]
Okay, last call. @georgschermann, @JjfBcn, @vmmaSigmaaie - if you don't object now, we will merge solution 2 tomorrow.

**Comment by georgschermann** [comment]
Solution 2 is also perfectly fine for us

**Comment by wrygiel** [comment]
Closed in #25.
